### ISSUE TITLE
PHE-Co-occurring-substance-misuse-and-mental-health-issues

### ISSUE THREAD

**Issue by mikeAdamss** [GSS-Cogs/family-disability#8]
- [x] Create Jenkins job (in Jenkins).
- [x] Create a Jenkinsfile.
- [x] Figure out landing page. #19 
- [ ] Create main.py and check scraper works.
- [x] Determine eventual Tidy data column names and add to `info.json`.
- [x] Map column names to codelists (existing and newly required), adding mapping to `info.json`.
- [x] Add definition to `reference/columns.csv`
- [x] Add newly required codelists (under `reference/codelists`) and add info to `reference/codelists-metadata.json`
- [x] Add component definitions to `reference/components.csv`
- [x] Write a transformation to Tidy CSV and output metadata `dataset.trig` and schema. 
     - [x]  Draft
     - [ ]  Finalised

**Comment by mikeAdamss** [comment]
blocked by https://github.com/GSS-Cogs/gss-utils/issues/13

**Comment by mikeAdamss** [comment]
Chatted with Alex and have removed the blocked label. Will build the transform against a download copy of the data and treat the scraper as a separate task.

**Comment by mikeAdamss** [comment]
Have some initial concerns about the data source, as follows:

I’m using the default .csv download as we don’t have a scraper for this yet, these are my concerns:

* The Category dimension has no values, at least on the default download.
* The Sex dimension has only values of “Persons” and “Not applicable” (I’ve no idea how that can be na).
* There’s a mix of partial and full years for time, i.e 2015 & 2015/2016 & 2016. Can we mix these?
* The data is 99.5% sparse (no idea if that matters for cogs cubes)

Highlighted for DM's to have a look at.

**Comment by grace-spitzer-wong** [comment]
Indicator 90371: 
- parent code/parent name duplicates Area code and Area name. 
- Area type is a duplicate and can be removed. 
- Sex is persons - has this been kept as that in the past or removed? 
- Lower and upper CIs, new data and compared to goal empty - remove

**Comment by mikeAdamss** [comment]
ok, so this is complicated. I've created a separate gitrepo with all the basic data for a profile and the metadata for each of the indicators here: [https://github.com/mikeAdamss/data_temp](https://github.com/mikeAdamss/data_temp). Hopefully that'll help with understanding what we're looking at.

the problem is that fingertips has no concept of "dataset" per se, with the whole system revolving around the idea of indicators (slices of other data sources) organised into groups ("Risk and Related Factors" etc) several of which are combined to make a profile ("/drugsAndMentalHealth" in this case). These indicators are not organised into a sensible whole in the way a dataset typically is, so there's some thought needed around how (and if) we can split the whole out.

**Splitting the Data: Options That I can See**
so there's three principle places you can split the data for a given fingertips "profile" (this spike represents the data for one profile).

indicators - an indication of what's being tracked/what's got better/worse.
domain - general umbrella the indicators fall under, i.e "Risk and Related Factors", "Mortality Indicators".
category - sexuality, religion, deprivation index. This one is a pain as category covers what would typically be 17 different dimensions.


**How I've split the data**

By category. I don't know what the _best_ (as in use experience) way to split the data is, but splitting out category appears to be necessary and that gets us to the 17 datasets I've put through the pipeline. We can split further but the number of datacubes goes up dramatically, we might be able to get that down with synthetic dimensions, not sure.

**Metadata**

There's a whole slew of metadata and it's associated with the indicators. For this spike I've taken unit of measure and measure type as the absolute minimum.

The complication here is that the metadata is _per indicator_ not per category, so with the current split, each dataset/cube has a mix of indicators with each having (potentially) a different last updated day and/or providence etc.

**Data**

There's a ton of things you could take, again (with some input from Leigh and Grace) I've gone for the minimum.

**Geography**

I've just gone with the default geographic area. There's other options, some of them may be better. Again, didn't get to it but probably worth investigating and/or taking everything at all levels.

**Codelists**

For this I've just created codelists for most dimensions. There is some api information for them so something a bit cleverer than that should be possible once we know what we want to take in terms of data.

**Sparsity**

These will be really sparse. Darren is happy with that and wants to know how it'll play on the front end. Part of the spike nature of this one.

**Scraper**

I'm generating the trig and schema ad-hoc, because what we want the scraper to get depends on how we define a dataset which depends on how we decide to split out the data. It won't be hard, it's just a waste of time until we pin that down.

**Comment by ajtucker** [comment]
Re: scraper. Do we still need a nicer way to manage the metadata for PHE datasets? Issue GSS-Cogs/gss-utils#13 is about a scraper, but do we need to split out metadata from the scraper more?

**Comment by mikeAdamss** [comment]
tbh I'd close it for now anyway, it's not a viable task until we decide what is "a dataset" in fingertips land, and that's gonna affect how we get both the data and the metadata (might not be a scraper at all, could just be a thingy you pass ID's too that get data or metadata or both).

don't know is the honest answer, but I'd shut that down until we've worked this issue through to the end, and build tasks off that.

**Comment by grace-spitzer-wong** [comment]
@mikeAdamss  - any chance this would be pushing dupes through onto PMD? I haven't clicked through to everything but it looks like there's two for each one?

**Comment by grace-spitzer-wong** [comment]
Reply: These are old failed loads that weren’t quite right so likely to be just a SPARQL query
